tests/deploy/test_scheduler_agent.py

```diff
@@ -1286,6 +1285,72 @@ def make_resources(
     assert agent.executor_manager.executors["agent3"].execute_count == 0
 
 
+async def test_skipped_for_dependencies_with_event_propagation(agent: TestAgent, make_resource_minimal):
+    """
+    Ensure that a resource that was skipped for its dependencies gets scheduled when they succeed
+    """
+
+    rid1 = ResourceIdStr("test::Resource[agent1,name=1]")
+    rid2 = ResourceIdStr("test::Resource[agent2,name=2]")
+
+    # make both agent's executors managed
+    executor1: ManagedExecutor = ManagedExecutor()
+    agent.executor_manager.register_managed_executor("agent1", executor1)
+
+    executor2: ManagedExecutor = ManagedExecutor()
+    agent.executor_manager.register_managed_executor("agent2", executor2)
+
+    resources = {
+        rid1: make_resource_minimal(rid=rid1, values={"value": "r1_value"}, requires=[]),
+        rid2: make_resource_minimal(rid=rid2, values={"value": "r2_value"}, requires=[rid1]),
+    }
+    await agent.scheduler._new_version(version=1, resources=resources, requires=make_requires(resources))
+
+    # We set rid1 to fail so that rid2 has a failing dependency
```

Comment:
You immediately set up the failure scenario, but you don't verify the assumptions of the start condition, i.e. that event propagation is disabled. I think you should first deploy both successfully, then trigger a deploy for rid1, asserting that rid2 does not get deployed (proving that no events are sent), and only then set up the failure scenario so that the final assertion becomes meaningful.